Assets/SEE/Net/Network.cs

```diff
@@ -288,33 +294,106 @@ private void Start()
+        /// <exception cref="ArgumentException">thrown if an option requiring a value does
+        /// not have one</exception>
+        private void ProcessCommandLineArguments()
         {
-#if UNITY_EDITOR
-            if (networkCommsLoggingEnabled)
+            string[] arguments = Environment.GetCommandLineArgs();
+
+            // Check command line arguments
+            // The first element in the array contains the file name of the executing program.
+            // If the file name is not available, the first element is equal to String.Empty.
+            for (int i = 1; i < arguments.Length; i++)
             {
-                NetworkComms.EnableLogging(new NetworkCommsLogger(minimalSeverity));
+                if (arguments[i] == portArgument)
+                {
+                    CheckArgumentValue(arguments, i, portArgument);
+                    ServerPort = Int32.Parse(arguments[i + 1]);
+                    i++;
+                }
+                else if (arguments[i] == passwordArgument)
+                {
+                    CheckArgumentValue(arguments, i, passwordArgument);
+                    RoomPassword = arguments[i + 1];
+                    i++;
+                }
+                else if (arguments[i] == domainArgument)
+                {
+                    CheckArgumentValue(arguments, i, domainArgument);
+                    BackendDomain = arguments[i + 1];
+                    i++;
+                }
+                else if (arguments[i] == serverIdArgument)
+                {
+                    CheckArgumentValue(arguments, i, serverIdArgument);
+                    ServerId = arguments[i + 1];
+                    i++;
+                }
+                else if (arguments[i] == launchAsServerArgument)
+                {
+                    StartServer(null);
+                }
+                else
+                {
+                    Debug.LogWarning($"Unknown command-line parameter {arguments[i]} will be ignored.\n");
+                }
             }
-            else
+
+            static void CheckArgumentValue(string[] arguments, int i, string argument)
             {
-                NetworkComms.DisableLogging();
+                if (i + 1 >= arguments.Length)
+                {
+                    throw new ArgumentException($"Argument value for {argument} is missing.");
+                }
             }
-#else
-                NetworkComms.DisableLogging();
-#endif
+        }
 
-            if (HostServer)
+        /// <summary>
+        /// The <see cref="ServerActionNetwork"/> component attached to the server game object.
+        /// </summary>
+        private static ServerActionNetwork serverNetwork;
+        /// <summary>
+        /// The gateway to the server.
+        /// </summary>
+        public static ServerActionNetwork ServerNetwork
+        {
+            get
             {
-                Server.Initialize();
+                if (serverNetwork == null)
```

Comment:
We can simplify this kind of code by using [`Lazy<T>`](https://learn.microsoft.com/en-us/dotnet/framework/performance/lazy-initialization#basic-lazy-initialization)

Reply:
Implemented this suggestion.